flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionManagerTest.java

```diff
@@ -60,4 +61,26 @@ public void testCreateViewForRegisteredPartition() throws Exception {
 		partitionManager.registerResultPartition(partition);
 		partitionManager.createSubpartitionView(partition.getPartitionId(), 0, new NoOpBufferAvailablityListener());
 	}
+
+	@Test
+	public void testExternallyManagedPartitionReleaseIfFlagDisabled() {
```

Comment:
I think it would be better to have a common utility method with a parameters indicating whether to release externally managed partitions.

Reply:
I'm not sure I follow, who would use this method?

Reply:
Just `testExternallyManagedPartitionReleaseIfFlagDisabled` and `testExternallyManagedPartitionReleaseIfFlagEnabled`, these two methods seem to have most logic in common, so I think we may have a method like

`
public void testExternallyManagedPartitionRelease(boolean releaseExternalManagedPartitions) {
		final ResultPartitionManager partitionManager = new ResultPartitionManager(releaseExternalManagedPartitions);
		final ResultPartition partition = new ResultPartitionBuilder().setIsExternallyManaged(true).build();

		partitionManager.registerResultPartition(partition);
		partitionManager.onConsumedPartition(partition);

		assertThat(partition.isReleased(), is(releaseExternalManagedPartitions));
	}
`
and let the two tests call this method with different parameters.

Reply:
addressed in #8654.